lib/fbe/unmask_repos.rb

```diff
@@ -37,21 +37,23 @@ def self.mask_to_regex(mask)
 
   def self.unmask_repos(options: $options, global: $global, loog: $loog)
     repos = []
+    octo = Fbe.octo(loog:, global:, options:)
     masks = (options.repositories || '').split(',')
     masks.reject { |m| m.start_with?('-') }.each do |mask|
       unless mask.include?('*')
         repos << mask
         next
       end
       re = Fbe.mask_to_regex(mask)
-      Fbe.octo(loog:, global:, options:).repositories(mask.split('/')[0]).each do |r|
+      octo.repositories(mask.split('/')[0]).each do |r|
         repos << r[:full_name] if re.match?(r[:full_name])
       end
     end
     masks.select { |m| m.start_with?('-') }.each do |mask|
       re = Fbe.mask_to_regex(mask[1..])
       repos.reject! { |r| re.match?(r) }
     end
+    repos.reject! { |repo| octo.repository(repo)[:archived] }
```

Comment:
@Yegorov you are making an extra HTTP request to GitHub API. Maybe it's better to use the JSON already retrieved a few lines earlier, in line 48?

Reply:
> Maybe it's better to use the JSON already retrieved a few lines earlier, in line 48?

@yegor256  I examined this case, but there may be a situation when an added repository without a mask is already in the archive

See this code: we added repos without make request to GitHub API:
```ruby
      unless mask.include?('*')
        repos << mask
        next
      end
``` 
Therefore, we should do an archive test at the very end